contrib/babelfishpg_tsql/sql/sys_functions.sql

```diff
@@ -2778,4 +2778,71 @@ $$
 LANGUAGE SQL IMMUTABLE PARALLEL RESTRICTED;
 
 CREATE OR REPLACE FUNCTION sys.language()
-RETURNS sys.NVARCHAR(128)  AS 'babelfishpg_tsql' LANGUAGE C;
\ No newline at end of file
+RETURNS sys.NVARCHAR(128)  AS 'babelfishpg_tsql' LANGUAGE C;
+
+CREATE OR REPLACE FUNCTION sys.INDEXPROPERTY(IN object_id INT, IN index_or_statistics_name sys.nvarchar(128), IN property varchar(128))
+RETURNS INT AS
+$BODY$
+DECLARE
+ret_val INT;
+BEGIN
+	index_or_statistics_name = LOWER(TRIM(index_or_statistics_name));
+	property = LOWER(TRIM(property));
+    SELECT INTO ret_val
+    CASE
+       
+        WHEN (SELECT CAST(type AS int) FROM sys.indexes i WHERE i.object_id = $1 AND i.name = $2 COLLATE "C") = 3 -- is XML index
```

Comment:
I would use sys.database_default instead of "C". 

Applicable to other places also.

Reply:
@Deepesh125 ^ ?

Reply:
If we are in PG dialect then you should be able to use "COLLATE sys.database_default" and if you are in tsql dialect then you should be able to use "COLLATE database_default"

Reply:
Thanks, managed to figure it out with your help. Done.